infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -139,6 +143,15 @@ public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
         return true;
     }
 
+    ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat("sequencer-%d").build();
+
+    ExecutorService executor = Executors.newSingleThreadExecutor(threadFactory);
```

Comment:
Can you add a shutdown method to gracefully terminate this executor service?

Reply:
Also, the NettyServerRouter thread pool registers the exception handler `handleUncaughtException`. Is there any way to add it here too to maintain the same functionality?

Reply:
Done.